File: C++/level2/가장_큰수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string str1, string str2){
    if(str1+str2 > str2+str1)
        return true;
    else
        return false;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> strings;

    while(!numbers.empty()){
        strings.push_back(to_string(numbers.back()));
        numbers.pop_back();
    }

    sort(strings.begin(), strings.end(), compare);

    for(int i=0; i<strings.size(); i++)
        answer = answer + strings[i];

    if(answer[0] == '0')
        return "0";

    return answer;
}

File: JAVA/level2/가장_큰수.java
import java.util.*;

class Solution {

    public String solution(int[] numbers) {
        String answer = "";
        String [] strings = new String[numbers.length];

        for(int i=0; i<numbers.length; i++)
            strings[i] = Integer.toString(numbers[i]);

        Arrays.sort(strings, new Comparator<String>() {
            @Override
            public int compare(String str1, String str2) {
                return (str1+str2).compareTo(str2+str1);
            }
        });

        for(int i=0; i<numbers.length; i++)
            answer = strings[i] + answer;

        if(answer.charAt(0) == '0')
            return "0";
        return answer;
    }
}
